### ISSUE TITLE
improved customizable NICON

### ISSUE THREAD

**Issue by dcornet** [GBeurier/nirs4all#7]
"Improved" version of decon dedicated to fintuning. The aim is to avoid mismatch between large initial stride and subsequent large kernel_size. All values are explicit. Normalization declaration is safer. To be used in ref_models

def customizable_nicon(input_shape, params={}, verbose=False):
    """
    Builds a robust custom CNN model with depthwise convolutions.

    Parameters:
        input_shape (tuple): Shape of the input data.
        params (dict): Dictionary of parameters for model configuration.
        verbose (bool): If True, print layer output shapes at each step.

    Returns:
        keras.Sequential: Compiled model.
    """
    # Récupération des paramètres avec valeurs par défaut safe
    spatial_dropout = params.get('spatial_dropout', 0.08)
    filters1 = params.get('filters1', 8)
    kernel_size1 = params.get('kernel_size1', 5)
    strides1 = params.get('strides1', 1)
    activation1 = params.get('activation1', "selu")
    
    dropout_rate = params.get('dropout_rate', 0.2)
    filters2 = params.get('filters2', 16)
    kernel_size2 = params.get('kernel_size2', 5)
    strides2 = params.get('strides2', 1)
    activation2 = params.get('activation2', "relu")
    
    normalization_method1 = params.get('normalization_method1', "BatchNormalization")
    filters3 = params.get('filters3', 32)
    kernel_size3 = params.get('kernel_size3', 3)
    strides3 = params.get('strides3', 1)
    activation3 = params.get('activation3', "elu")
    normalization_method2 = params.get('normalization_method2', "BatchNormalization")
    
    dense_units = params.get('dense_units', 16)
    dense_activation = params.get('dense_activation', "sigmoid")
    final_activation = params.get('final_activation', "linear")  # ou "sigmoid" selon la tâche

    model = Sequential()
    model.add(Input(shape=input_shape))
    if verbose:
        print(f"Input shape: {input_shape}")
    model.add(SpatialDropout1D(spatial_dropout))
    model.add(Conv1D(filters=filters1, kernel_size=kernel_size1, strides=strides1, activation=activation1))
    if verbose:
        print(f"After Conv1D-1: {model.layers[-1].output_shape}")
    model.add(Dropout(dropout_rate))
    model.add(Conv1D(filters=filters2, kernel_size=kernel_size2, strides=strides2, activation=activation2))
    if verbose:
        print(f"After Conv1D-2: {model.layers[-1].output_shape}")
    if normalization_method1 == "BatchNormalization":
        model.add(BatchNormalization())
    else:
        model.add(LayerNormalization())
    model.add(Conv1D(filters=filters3, kernel_size=kernel_size3, strides=strides3, activation=activation3))
    if verbose:
        print(f"After Conv1D-3: {model.layers[-1].output_shape}")
    if normalization_method2 == "BatchNormalization":
        model.add(BatchNormalization())
    else:
        model.add(LayerNormalization())
    model.add(Flatten())
    model.add(Dense(dense_units, activation=dense_activation))
    model.add(Dense(1, activation=final_activation))
    if verbose:
        print("Model built successfully!")
    return model
